File: binary-search-tree-to-greater-sum-tree/binary-search-tree-to-greater-sum-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sum = 0;
    public TreeNode bstToGst(TreeNode root) {
            rwsol(root);
            return root;    
    }
    public void rwsol(TreeNode root)
    {
        if(root == null)
        {
            return;
        }
        rwsol(root.right);
        int od = root.val;
        root.val = sum+root.val;
        sum+=od;
        rwsol(root.left);
    }
}

File: binary-search-tree-to-greater-sum-tree/binary-search-tree-to-greater-sum-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
sm = 0;
def solve(root):
    global sm
    if root:
        solve(root.right)
        od = root.val;
        root.val = sm+root.val;
        sm+=od;
        solve(root.left);
    return;
class Solution(object):
    def bstToGst(self, root):
        solve(root);
        global sm
        sm = 0;
        return root;
        
        